hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -312,17 +380,37 @@ public static List<UnhealthyContainers> generateUnhealthyRecords(
      */
     public static List<UnhealthyContainers> generateUnhealthyRecords(
         ContainerHealthStatus container, Set<String> recordForStateExists,
-        long time) {
+        long time,
+        Map<String, Map<String, Long>> unhealthyContainerStateStatsMap) {
       List<UnhealthyContainers> records = new ArrayList<>();
       if (container.isHealthy() || container.isDeleted()) {
         return records;
       }
 
       if (container.isMissing()
           && !recordForStateExists.contains(
-              UnHealthyContainerStates.MISSING.toString())) {
-        records.add(
-            recordForState(container, UnHealthyContainerStates.MISSING, time));
+          UnHealthyContainerStates.MISSING.toString())) {
+        if (!container.isEmpty()) {
+          LOG.info("Container is missing but not empty, mapped with {} " +
+                  "number of keys and having {} usedBytes in SCM metadata. " +
+                  "For all OM keys mapped to this container, pls check the " +
+                  "missing container page in Recon which shows all the keys " +
+                  "and their metadata.", container.getNumKeys(),
+              container.getContainer().getUsedBytes());
+          records.add(
+              recordForState(container, UnHealthyContainerStates.MISSING,
+                  time));
+          populateContainerStats(container,
+              UnHealthyContainerStates.MISSING.toString(),
+              unhealthyContainerStateStatsMap);
+        } else {
+          records.add(
+              recordForState(container, UnHealthyContainerStates.EMPTY_MISSING,
+                  time));
```

Comment:
Consider adding a debug level logging above this line?

Reply:
Done